src/OrchardCore.Modules/OrchardCore.Tenants/Workflows/Activities/CreateTenantTask.cs

```diff
@@ -0,0 +1,114 @@
+        }
+
+        public WorkflowExpression<string> ConnectionString
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> TablePrefix
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> RecipeName
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+        
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return Outcomes(T["Done"]);
+        }
+
+        public async override Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            var tenantNameTask = _expressionEvaluator.EvaluateAsync(TenantName, workflowContext);
+            var requestUrlPrefixTask = _expressionEvaluator.EvaluateAsync(RequestUrlPrefix, workflowContext);
+            var requestUrlHostTask = _expressionEvaluator.EvaluateAsync(RequestUrlHost, workflowContext);
+            var databaseProviderTask = _expressionEvaluator.EvaluateAsync(DatabaseProvider, workflowContext);
+            var connectionStringTask = _expressionEvaluator.EvaluateAsync(ConnectionString, workflowContext);
+            var tablePrefixTask = _expressionEvaluator.EvaluateAsync(TablePrefix, workflowContext);
+            var recipeNameTask = _expressionEvaluator.EvaluateAsync(RecipeName, workflowContext);
+
+            await Task.WhenAll(tenantNameTask, requestUrlPrefixTask, requestUrlHostTask, databaseProviderTask, connectionStringTask, tablePrefixTask, recipeNameTask);
+
+            var shellSettings = new ShellSettings();
+
+            if (!string.IsNullOrWhiteSpace(tenantNameTask.Result))
+            {
+                shellSettings = new ShellSettings
+                {
+                    Name = tenantNameTask.Result?.Trim(),
+                    RequestUrlPrefix = requestUrlPrefixTask.Result?.Trim(),
+                    RequestUrlHost = requestUrlHostTask.Result?.Trim(),
+                    State = TenantState.Uninitialized
+                };
+                shellSettings["ConnectionString"] = connectionStringTask.Result?.Trim();
+                shellSettings["TablePrefix"] = tablePrefixTask.Result?.Trim();
+                shellSettings["DatabaseProvider"] = databaseProviderTask.Result?.Trim();
+                shellSettings["Secret"] = Guid.NewGuid().ToString();
+                shellSettings["RecipeName"] = recipeNameTask.Result.Trim();
+            }
+            
+            ShellSettingsManager.SaveSettings(shellSettings);
+            var shellContext = await ShellHost.GetOrCreateShellContextAsync(shellSettings);
```

Comment:
i will have to change it to `await _shellHost.UpdateShellSettingsAsync(shellSettings)` in my #4164 PR, so that configs / settings are really re-loaded from the related sources, e.g if data have not been well persisted, if we are using shared configs / settings sources in a distributed env ...

Reply:
and if no task name you still save the settings